pkg/util/files.go

```diff
@@ -60,3 +64,44 @@ func createDirIfNotExists(path string) error {
 	}
 	return nil
 }
+
+// ReadFileSource reads the file source which is either relative to the k3d config file or embedded in it.
+func ReadFileSource(configFile, sourcePath, sourceContent string) []byte {
+	if sourcePath != "" {
+		sourceFilePath := filepath.Join(filepath.Dir(configFile), sourcePath)
+		fileInfo, err := os.Stat(sourceFilePath)
+		if err == nil && !fileInfo.IsDir() {
+			content, err := os.ReadFile(sourceFilePath)
+			if err != nil {
+				l.Log().Warnf("Cannot read file: %s", sourceFilePath)
+				return nil
+			}
+			return content
+		}
+		l.Log().Warnf("Could resolve source file path: %s", sourceFilePath)
```

Comment:
@iwilltry42 Should k3d fail if it can't find the files or just print a warning? Since it's not a fatal error, I just print a warning now. But it's up to you if it should exit with an error.

Reply:
I would think it's a fatal error though, since we should assume that whatever is in the config file is crucial for the cluster to be considered ready.
I.e. if I put my CNI config in there, I'd expect cluster creation to fail if it cannot be deployed (apart from errors in whatever is deployed obviously.)

Reply:
I would think it's a fatal error though, since we should assume that whatever is in the config file is crucial for the cluster to be considered ready.
I.e. if I put my CNI config in there, I'd expect cluster creation to fail if it cannot be deployed (apart from errors in whatever is deployed obviously.)